src/modal/modal.spec.ts

```diff
@@ -587,6 +587,45 @@ describe('ngb-modal', () => {
       fixture.detectChanges();
     });
   });
+
+  describe('accessibility', () => {
+    function testAttributes(modalOptions, expectedAttributes) {
+      const modalInstance = fixture.componentInstance.open('foo', modalOptions);
+      fixture.detectChanges();
+
+      const modalElement = <HTMLElement>document.querySelector('ngb-modal-window');
+      expectedAttributes.forEach(([name, value]) => expect(modalElement.getAttribute(name)).toBe(value));
+
+      modalInstance.close('some result');
+      fixture.detectChanges();
+      expect(fixture.nativeElement).not.toHaveModal();
+    }
+
+    it('should support aria-label', () => {
+      const label = 'This is the aria-label';
+
+      testAttributes({ariaLabel: label}, [['aria-label', label]]);
+    });
+
+    it('should support aria-labelledby', () => {
+      const id = 'aria-labelledby-id';
+
+      testAttributes({ariaLabelledBy: id}, [['aria-labelledby', id]]);
+    });
+
+    it('should make aria-label have priority over aria-labelledby', () => {
```

Comment:
`aria-labelledby` should have priority over `aria-label`.

See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute

> When user agents compute the accessible name property of elements that have both an aria-labelledby attribute and an aria-label attribute, the user agents give precedence to aria-labelledby.

Reply:
That's actually what I implemented the first time, but I didnd't keep the logic for two reasons: 

- it's already done by the browser as you mentioned
- I prefer to keep the logic here focused on displaying rather than trying to guess what should be expected; let me explain below

If the user wants, for any reason, to have both `aria-label` and `aria-labelledby` present in the DOM, we shouldn't prevent him from doing so. Moreover, `aria-label` could be used as a fallback in case the id specified in `aria-labelledby` doesn't exist.

Reply:
Wait @bastienmoulia I just noticed I made a mistake and my branch is wrong, let me change it.